lib/src/Base/Func/SpecFunc/SpecFunc.cxx

```diff
@@ -1135,6 +1135,29 @@ UnsignedInteger SpecFunc::NextPowerOfTwo(const UnsignedInteger n)
   return powerOfTwo;
 }
 
+// Integer power
+Scalar SpecFunc::IPow(const Scalar x, const SignedInteger n)
+{
+  if (n == 0) return 1.0;
+  if (x == 0) return 0.0;
+  if (x < 0.0) return -std::pow(-x, 1.0 * n);
```

Comment:
what should be the output of `IPow(-5.0, 2)`? I think you should check `n%2`, isn't it?

Reply:
Right! I will fix it ASAP. Concerning the documentation, my first intent was to provide a fix for the issue in the C++ code, to be cherry-picked and completed (doc+test) by someone else as there were already two PRs on this issue. But I propose to be the one in charge of solving #1515, including code from the other PRs and adding test&doc to this PR.

Reply:
Concerning the doc it is more general. As the class is public, we should provide some docstrings for **all existing methods/attributes**

I can start soon with *attributes* and some methods